app/services/enqueue_index_jobs.rb

```diff
@@ -44,7 +44,7 @@ def stats
   end
 
   def index_states
-    @index_states ||= BookIndexState.live
+    @index_states ||= BookIndexState.live.to_a
```

Comment:
Changing index_states from a lazy loader to an array fixes the issue with in_demand being lost.